chunks 1,5: Do not set date to UNDATED (to be pushed upstream).
chunk 2: Tweak page headers (local change).
chunk 3: Put architecture name in parentheses (to be pushed upstream).
chunk 4: Default value for the .Os macro (local change).
chunk 6: Do not use an ugly double dash in the NAME line (local change).
chunk 7: Set the base paragraph indentation to 5n (upstream commit 5d2e49f8).

Index: tmac/mdoc/doc-common
--- tmac/mdoc/doc-common.orig
+++ tmac/mdoc/doc-common
@@ -278,7 +278,7 @@
 .\" NS doc-date-string global string
 .\" NS   the manual page date as set by 'Dd'
 .
-.ds doc-date-string UNDATED\"
+.ds doc-date-string \" empty
 .
 .
 .\" NS Dd user macro (not parsed, not callable)
@@ -335,7 +335,7 @@
 .ds doc-volume-ds-1 General Commands Manual
 .ds doc-volume-ds-2 System Calls Manual
 .ds doc-volume-ds-3 Library Functions Manual
-.ds doc-volume-ds-4 Kernel Interfaces Manual
+.ds doc-volume-ds-4 Device Drivers Manual
 .ds doc-volume-ds-5 File Formats Manual
 .ds doc-volume-ds-6 Games Manual
 .ds doc-volume-ds-7 Miscellaneous Information Manual
@@ -455,12 +455,11 @@
 .    ds doc-section \$2
 .    ie \B'\$2' \{\
 .      if ((\$2 >= 1) & (\$2 <= 9)) \{\
-.        ds doc-volume \" empty (not "LOCAL")
+.        ds doc-volume "\*[doc-volume-ds-\$2]
 .        if \A'\$3' \{\
 .          if d doc-volume-as-\$3 \
-.            as doc-volume "\*[doc-volume-as-\$3]
-.          \}
-.        as doc-volume " \*[doc-volume-ds-\$2]
+.            as doc-volume " (\*[doc-volume-as-\$3])
+.        \}
 .      \}
 .    \}
 .    el \{\
@@ -490,7 +489,7 @@
 .\" NS
 .\" NS override this in 'mdoc.local', if necessary
 .
-.ds doc-default-operating-system GNU\"
+.ds doc-default-operating-system OpenBSD\"
 .
 .
 .\" NS doc-operating-system global string
@@ -1150,7 +1149,7 @@
 .  doc-break-page-with-new-number
 .
 .  \" Reset strings to reduce info leaks from one man page to the next.
-.  ds doc-date-string UNDATED\"
+.  ds doc-date-string \" empty
 .  ds doc-page-topic UNTITLED\"
 .  ds doc-volume LOCAL\"
 .  ds doc-section \" empty
@@ -1244,7 +1243,7 @@
 .
 .eo
 .de Nd
-.  nop \[em] \$*
+.  nop \[en] \$*
 ..
 .ec
 .
@@ -1482,7 +1481,7 @@
 .    if t \
 .      ss \n[doc-reg-Sh] \n[doc-reg-Sh1]
 .
-.    in \n[IN]u
+.    in \n[BP]u
 .    ns
 .
 .    doc-check-depth
